### PR TITLE
Support GCP CloudSQL proxy

### LINKED ISSUE #6562: Support GCP CloudSQL

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Description

Support GCP CloudSQL. Also change the config so the external DB is provider-agnostic

## Related Issue(s)

Fixes #6562 

## How to test

Set up MySQL DB in GCP

Create secret with following keys:
- `credentials.json` - service account to access DB
- `encryptionKeys` - use `[{"name": "general","version": 1,"primary": true,"material": "4uGh1q8y2DYryJwrVMHs0kWXJlqvHWWt/KJuNi04edI="}]`
- `username` - whatever you set up in GCP
- `password` - ditto

Change config to:
```
database:
  cloudSQL:
    serviceAccount:
      kind: secret
      name: <secretName>
    instance: <projected>:<region>:<dbName>
```

## Release Notes

```release-note
Add support for GCP CloudSQL
```

### CODE REVIEW

**Comment by roboquat** [comment]
@MrSimonEmms: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **mrsimonemms** after the PR has been reviewed.

Associated issue: *#6562*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: installer/pkg/components/database/cloudsql/init/00-create-and-init-sessions-db.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+-- Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+-- Licensed under the MIT License. See License-MIT.txt in the project root for license information.
+
+-- must be idempotent
+
+CREATE DATABASE IF NOT EXISTS `gitpod-sessions` CHARSET utf8mb4;
```
Comment (csweichel) [inline installer/pkg/components/database/cloudsql/init/00-create-and-init-sessions-db.sql:6]: We could - not necessarily in this PR - move this script (these scripts) to something like `pkg/components/database/init`

**Review by csweichel — COMMENTED** [review]
It would also be cool to have a secret validation in place akin to the `certificates` check.

That said, both comments in this PR can come in later and don't need to be part of this PR. I'd be happy to merge this way (just remove the hold), or add to this PR.

/hold
/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: ed4a33285ff24abfadf954dd2a0c7b267ed75a55</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6606#pullrequestreview-801163968" title="LGTM">csweichel</a>*

Associated issue: *#6562*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: installer/pkg/components/database/cloudsql/init/00-create-and-init-sessions-db.sql (hunk 1 above)
Comment (MrSimonEmms) [reply installer/pkg/components/database/cloudsql/init/00-create-and-init-sessions-db.sql:6]: Possible, but the init scripts are subtly different.

**Pull request merged**